lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py

```diff
@@ -0,0 +1,247 @@
+              needs to be at least 2 items in the list
+        type: list
+author:
+    - "Rui Moreira (@ruimoreira)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+- dms_replication_subnet_group:
+    state: present
+    identifier: "dev-sngroup"
+    description: "Development Subnet Group asdasdas"
+    subnet_ids: ['subnet-id1','subnet-id2']
+'''
+
+RETURN = ''' # '''
+__metaclass__ = type
+import traceback
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, HAS_BOTO3, \
+    camel_dict_to_snake_dict, get_aws_connection_info, AWSRetry
+try:
+    import botocore
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+backoff_params = dict(tries=5, delay=1, backoff=1.5)
+
+
+@AWSRetry.backoff(**backoff_params)
+def describe_subnet_group(connection, subnet_group):
+    """checks if instance exists"""
+    try:
+        subnet_group_filter = dict(Name='replication-subnet-group-id',
+                                   Values=[subnet_group])
+        return connection.describe_replication_subnet_groups(Filters=[subnet_group_filter])
+    except botocore.exceptions.ClientError:
+        return {'ReplicationSubnetGroups': []}
+
+
+@AWSRetry.backoff(**backoff_params)
+def replication_subnet_group_create(connection, **params):
+    """ creates the replication subnet group """
+    return connection.create_replication_subnet_group(**params)
+
+
+@AWSRetry.backoff(**backoff_params)
+def replication_subnet_group_modify(connection, **modify_params):
+    return connection.modify_replication_subnet_group(**modify_params)
+
+
+@AWSRetry.backoff(**backoff_params)
+def replication_subnet_group_delete(module, connection):
+    subnetid = module.params.get('identifier')
+    delete_parameters = dict(ReplicationSubnetGroupIdentifier=subnetid)
+    return connection.delete_replication_subnet_group(**delete_parameters)
+
+
+def get_dms_client(module, aws_connect_params, client_region, ec2_url):
+    client_params = dict(
+        module=module,
+        conn_type='client',
+        resource='dms',
+        region=client_region,
+        endpoint=ec2_url,
+        **aws_connect_params
+    )
+    return boto3_conn(**client_params)
+
+
+def replication_subnet_exists(subnet):
+    """ Returns boolean based on the existance of the endpoint
+    :param endpoint: dict containing the described endpoint
+    :return: bool
+    """
+    return bool(len(subnet['ReplicationSubnetGroups']))
+
+
+def create_module_params(module):
+    """
+    Reads the module parameters and returns a dict
+    :return: dict
+    """
+    instance_parameters = dict(
+        # ReplicationSubnetGroupIdentifier gets translated to lower case anyway by the API
+        ReplicationSubnetGroupIdentifier=module.params.get('identifier').lower(),
+        ReplicationSubnetGroupDescription=module.params.get('description'),
+        subnet_ids=module.params.get('subnet_ids'),
```

Comment:
```suggestion
        SubnetIds=module.params.get('subnet_ids'),
```
Here this var needs to stay as you had it originally - ansible prefers snake cased but boto typically needs camelcase.  `instance_parameters` will be passed into the boto connection so needs to match what the API expects, both here and later when you access the returned parameters.
https://docs.aws.amazon.com/dms/latest/APIReference/API_CreateReplicationSubnetGroup.html#API_CreateReplicationSubnetGroup_RequestSyntax